### PR TITLE
Compute commands: support for Diagnostics extension, bug fixes, tests refactoring and updates

### PR DESCRIPTION
**Pull request by ruslangabitov**
This PR contains the following changes:

1. Added support for IAAS diag and Linux diag extensions.
2. Fixed compute bugs.
3. Added test cases for IAAS diagnostics extension.
4. Added test cases to address bug fixes
5. Updated recordings with minor refactoring

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ruslangabitov__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: lib/commands/arm/vm/vmImage._js [hunk 1]
```diff
@@ -104,136 +104,191 @@ __.extend(VMImage.prototype, {
...
-    if (imageFilter.publishername && imageFilter.offer) {
-      this.cli.output.info('The parameter --sku if specified will be ignored');
-      skusCollection = this.getVMImageSkusList(imageFilter.location, imageFilter.publishername, imageFilter.offer, _);
-      imageQueries = [];
-      skusCollection.resources.forEach( function(skus) {
+  _getVMImageListForPublisherAndOffer: function (imageFilter, _) {
+    var that = this;
+    var imageQueries;
+
+    this.cli.output.info('The parameter --sku if specified will be ignored');
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmImage._js]: nit, once you have assigned 'this' to 'that', please use 'that' across

File: lib/commands/arm/vm/vmImage._js [hunk 2]
```diff
@@ -104,136 +104,191 @@ __.extend(VMImage.prototype, {
...
+    var skusCollection = this.getVMImageSkusList(imageFilter.location, imageFilter.publishername, imageFilter.offer, _);
+    imageQueries = [];
+    skusCollection.resources.forEach( function(skus) {
+      imageQueries.push(function(callBack) {
+        skus = skus;
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmImage._js]: this appears redundant to me.

**Comment by yugangw-msft** [comment]
LGTM overall. Will merge after you clarify the couple of review comment

**Pull request merged**